src/conversion/viewconversiondispatcher.js

```diff
@@ -137,55 +157,68 @@ export default class ViewConversionDispatcher {
 	 * @fires documentFragment
 	 * @param {module:engine/view/documentfragment~DocumentFragment|module:engine/view/element~Element} viewItem
 	 * Part of the view to be converted.
-	 * @param {Object} additionalData Additional data to be passed in `data` argument when firing `ViewConversionDispatcher`
-	 * events. See also {@link ~ViewConversionDispatcher#event:element element event}.
+	 * @param {module:engine/model/schema~SchemaContextDefinition} [context=['$root']] Elements will be converted according to this context.
 	 * @returns {module:engine/model/documentfragment~DocumentFragment} Model data that is a result of the conversion process
 	 * wrapped in `DocumentFragment`. Converted marker elements will be set as that document fragment's
 	 * {@link module:engine/model/documentfragment~DocumentFragment#markers static markers map}.
 	 */
-	convert( viewItem, additionalData ) {
+	convert( viewItem, context = [ '$root' ] ) {
 		return this._model.change( writer => {
+			this.fire( 'viewCleanup', viewItem );
+
+			const consumable = ViewConsumable.createFrom( viewItem );
+
+			// Create context tree and store position in the top element.
+			// Items will be converted according to this position.
+			this._contextPosition = contextToPosition( context, writer );
```

Comment:
`contextToPosition` -> `createContextTree`? This method is more about creating than type chage.

Reply:
and `_contextPosition` -> `modelCursor`?

Reply:
👍 I like it.